helix-core/src/main/java/org/apache/helix/controller/stages/StateTransitionThrottleController.java

```diff
@@ -99,6 +99,21 @@ public StateTransitionThrottleController(Set<String> resources, ClusterConfig cl
     }
   }
 
+  // The getter methods are used for debugging and testing purpose
```

Comment:
I am not 100% comfortable with adding methods just for this purpose. Here's a trick you can do (I have done this in some tests):

Instead of adding methods (that shouldn't otherwise be used other than testing purposes), create a child class (for example, `MockStateTranisitonThrottleController`) that extends this class and just add these test/debug methods. That way, in your unit tests, you could use the mock class and still utilize these methods that expose internal variables, rather than adding methods directly to the main class.

Do you think that would work?

Reply:
In addition to the subclass, you also need to make the fields protected otherwise they won't be accessible from subclass either. Making it protected doesn't make things better versus package-private method. Do you know why we need to have getMethods instead of directly accessing the internal variables? Because you could implement get methods in a more self defensive manner but `protected` just exposes the private field

Reply:
Yes, you would need to make the relevant fields `protected`. That was implied. `protected` should be good enough as long as we don't make it `public`. The point here is to avoid adding methods that aren't strictly part of the class logic :)